### ISSUE TITLE
Paths to files in extensions are not resolved in TYPO3 12.4

### ISSUE THREAD

**Issue by akiessling** [bmack/modern-template-building#5]
https://github.com/bmack/modern-template-building/blob/1415d17f911f472aae1d37f478597b97aaf3fb2d/Classes/ContentObject/FileContentObject.php#L37

I'm not sure if the sanitize method behaves in TYPO3 12.4 like before. I can't load templates files from extensions due to this:

When i hand over a file like EXT:site/foo/bar.html it needs the second param $allowExtensionPath set to true, otherwise it tries to make a relative path and throws an exception that is only allowed for public files.
But if it is set to true it returns EXT:site/foo/bar.html and not vendor/my/site/foo/bar.html

```
        // this call also resolves EXT:myext/ files
        $absolutePath = GeneralUtility::getFileAbsFileName($file);
        if (!$absolutePath || is_dir($absolutePath)) {
            throw new FileDoesNotExistException('File "' . $file . '" was not found', 1530169845);
        }

        $isExtensionPath = PathUtility::isExtensionPath($file);
        if ($allowExtensionPath && $isExtensionPath) {
            return $file;
        }
```

Thus if $file is returned with EXT: we need to make it an absolute path again
```
$file = GeneralUtility::getFileAbsFileName($file);
```

**Comment by bmack** [comment]
Hey @akiessling ,

can you create a PR?

**Comment by akiessling** [comment]
Hey @bmack,

sorry, i was not working on the project for a couple of days. I only tested the fixes in my 12.4 update project with PHP 8.3, thus the additional fix
